authorization/src/main/java/life/qbic/authorization/application/policy/directive/InformUserAboutGrantedAccess.java

```diff
@@ -0,0 +1,77 @@
+package life.qbic.authorization.application.policy.directive;
+
+import java.util.Objects;
+import life.qbic.authentication.domain.user.concept.User;
+import life.qbic.authentication.domain.user.concept.UserId;
+import life.qbic.authentication.domain.user.repository.UserRepository;
+import life.qbic.authorization.application.AppContextProvider;
+import life.qbic.domain.concepts.DomainEvent;
+import life.qbic.domain.concepts.DomainEventSubscriber;
+import life.qbic.domain.concepts.communication.EmailService;
+import life.qbic.projectmanagement.domain.project.service.event.ProjectAccessGranted;
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.scheduling.JobScheduler;
+import org.springframework.stereotype.Component;
+
+/**
+ * <b>Directive: Inform user about granted access</b>
+ * <p>
+ * Notifies the user via email about the recently granted project access.
+ *
+ * @since 1.0.0
+ */
+@Component
+public class InformUserAboutGrantedAccess implements DomainEventSubscriber<ProjectAccessGranted> {
+
+  private final EmailService emailService;
+
+  private final JobScheduler jobScheduler;
+  private final UserRepository userRepository;
```

Comment:
I think this would be a strong violation of the Single Responsibility Pattern. I think it is not of the user contact service concern to know what it is about, but to deliver a message to a user with a given id. Imo, the service should care about finding the user's contact email address and sending the mail and only this. The content comes from the domain, where business logic policy demand to inform the user about an event.